tests/ssl-opt.sh

```diff
@@ -876,6 +876,38 @@ is_gnutls() {
     esac
 }
 
+# Determine what calc_verify trace is to be expected, if any.
+#
+# calc_verify is only called for two things: to calculate the
+# extended master secret, and to process client authentication.
+#
+# Warning: the current implementation assumes that extended_ms is not
+#          disabled on the client or on the server.
+#
+# Inputs:
+# * $1: 'required' or 1 if client authentication is expected,
+#       'none' or 0 or absent if not.
```

Comment:
I find 'none' a bit unintuitive, in that it will still end up setting the variable to something non-empty it if EMS is enabled. I'm not sure I have a better name to suggest, though. Perhaps 'auto'?

Reply:
Originally this was the `auth_mode` parameter. I ended up never calling this function with a parsed `auth_mode` (and in fact only ever calling it with `none`), but that could happen so I've kept the code but documented it.